docs/specs/add-kubernetes-support.md

```diff
@@ -0,0 +1,247 @@
++			},
++			RestartPolicy:      v1.RestartPolicyNever,
++			ServiceAccountName: serviceAccountName,
++		},
++	}
++
++       k8scli.CoreV1().Pods(ns).Create(pod)
++
++ }
+
+```
+
+If there are identical API objects between COEs, check which cluster is
+being used.
+```diff
++ func CreateClusterRoleBinding(roleBindingName string, svcAccountName string,
++                               ApbRole string, ns string) {
++	roleBindingM := map[string]interface{}{
++		"apiVersion": "v1",
++		"kind":       "RoleBinding",
++		"metadata": map[string]string{
++			"name":      roleBindingName,
++			"namespace": ns,
++		},
++		"subjects": []map[string]string{
++			map[string]string{
++				"kind":      "ServiceAccount",
++				"name":      svcAccountName,
++				"namespace": ns,
++			},
++		},
++		"roleRef": map[string]string{
++			"name": ApbRole,
++		},
++      }
++
++      if OpenShift {
++         OCcli.CoreV1().ClusterRoleBinding(ns).Create(roleBindingM)
++      } elif Kubernetes {
++         k8scli.CoreV1().ClusterRoleBinding(ns).Create(roleBindingM)
++      }
++ }
+
+```
+
+### Hierarchy Diagram
+The diagram represents the code split into two files and the API objects
+in each of them
+
+**pkg/cluster/kubernetes.go**
+
+| API Resource | Uses API | Uses CLI | Different between OpenShift and Kubernetes |
+|:---:|:---:|:---:|:---:|
+| Pods | X |  |  |
+| Exec |  | X |  |
+| Login |  | X |  |
+| ServiceAccount |  | X |  |
+| RBAC ClusterRoleBindings |  | X | X |
+| RBAC ClusterRoles |  | X | X |
+| Namespace |  | X | X |
+
+**pkg/cluster/openshift.go**
+
+| API Resource | Uses API | Uses CLI | Different between OpenShift and Kubernetes |
+|:---:|:---:|:---:|:---:|
+| ClusterRoleBindings |  | X | X |
+| ClusterRoles |  | X | X |
+| Project |  | X | X |
+
+## APBs
+The APBs only target OpenShift as a cluster. The APBs will differ in two ways:
+ - Packages
+ - oc-login.sh
+
+### Packages
+The main issue with packaging is that folks using Kubernetes may only want the
+Kubernetes client. The package ```origin-clients``` installs both the OpenShift
+and Kubernetes clients.  If this the container size isn't an issue, then there
+may not be a problem using ```origin-clients```.
+
+### Script
+The ```oc-login.sh``` script should be changed to accept an environment varible
+identifing the $COE.  Based on $COE, the login will use the ```kubectl``` or
+```oc``` client.
+
+## Tools and Docs that reference the OpenShift client
+**Tools**
+
+ - scripts/prep_local_devel_env.sh
+ - scripts/broker-ci/setup.sh
+ - scripts/run_latest_build.sh
+ - scripts/deploy.sh
+ - scripts/broker-ci/wait-for-pods.sh
+ - .travis.yml
+
+**Docs**
+
+ - docs/bind-data-transmission.md
+ - docs/local_development.md
+ - README.md
+
+## CI
+The community has two tools for CI: Jenkins and Travis. When adding
+Kubernetes support to the broker, the upstream Travis CI job should test the
+broker and catalog on top of the upstream COE, Kubernetes.  Jenkins will become
+the downstream CI testing on OpenShift.
```

Comment:
Do we need to have Travis only test Kubernetes support?

If we could also do a sanity check with oc cluster up like we do now that'd be good feedback to have with a PR.  

Essentially have the gate checking both K8S and OCP behavior.